apps/passport-client/src/service-worker.ts

```diff
@@ -1,65 +1,307 @@
 // this file is loaded as a service worker
 
+// Hack to make TypeScript aware of the ServiceWorkerGlobalScope type.
+// We can't redeclare `self`, but `swSelf` can be used with the right type.
+/// <reference lib="webworker" />
+const swSelf = self as unknown as ServiceWorkerGlobalScope;
+
+// Global flag to enable or disable service worker quickly.
+export const SERVICE_WORKER_ENABLED =
+  true || process.env.NODE_ENV !== "development"; // TODO(artwyman): cleanup before checkin
+
 /**
- * These files are cleared from the cache every time a new version
- * of the service worker is installed, because a new version of
- * the service worker is triggered by a change in the service worker
- * source code, which happens each time production deploys.
+ * Ephemeral cache is for resources which may change frequently, where we
+ * want to optimize for freshness, but allow caching for offline.  This cache
+ * is also updated dynamically as resources are fetched.  Cache name includes
+ * deployed code version, which protects against using
+ * cached pages which came from a different server deploy.
  */
-const impermanentCache = [
-  // "/",
-  // "/favicon.ico",
-  // "/index.html",
-  // "/global-pcdpass.css",
-  // "/global-zupass.css",
-  // "/js/index.js",
+const EPHEMERAL_CACHE_NAME = `v1-${process.env.SW_ID}`;
```

Comment:
`process.env.SW_ID` changes on every deploy. what are the cases you imagine we would change `v1-` to another string?

Reply:
Philosophically, I think we'd change it if we altered how we were using the cache in an incompatible way.  The potential reasons to do so are pretty limited, since the cache doesn't really have a format which can change.

Practically, we need to change it when we want to fully claer the stable cache.  I think it's unnecessary in the ephemeral cache name except as a way of marking them as being related to each other.

I'll split this out into its own string, and explain when it should change.  I think I'll also set it to v2 since we're certainly not using the cache in the same way as it was.